examples/js/ViveController.js

```diff
@@ -1,47 +1,95 @@
 /**
  * @author mrdoob / http://mrdoob.com
+ * @author stewdio / http://stewd.io
  */
 
-THREE.ViveController = function ( id ) {
+THREE.ViveController = function( id ) {
 
 	THREE.Object3D.call( this );
 
-	var scope = this;
-	var gamepad;
+	var gamepad, scope = this;
 
-	this.getGamepad = function () { return gamepad; };
+	this.getGamepad = function() {
+
+		return gamepad;
+
+	}
 	this.matrixAutoUpdate = false;
 	this.standingMatrix = new THREE.Matrix4();
+	this.axes = [ 0, 0 ];
+	this.thumbpadIsPressed = false;
+	this.triggerIsPressed = false;
+	this.gripsArePressed = false;
+	this.menuIsPressed = false;
```

Comment:
I don’t have good reasons for the first two (`matrixAutoUpdate` and `standingMatrix`)... Happy to have them as private.

The rest I make public so that in your main animation loop you can more easily reason about things while a button is pressed—for example: 
`if( controller.triggerIsPressed ){ continue shooting at the monster! }`